docs/source/weights_loading.rst

```diff
@@ -92,37 +88,42 @@ You can manually save checkpoints and restore your model from the checkpointed s
 Checkpoint Loading
 ------------------
 
-To load a model along with its weights, biases and hyperparameters use following method.
+To load a model along with its weights, biases and `module_arguments` use following method.
```

Comment:
@Borda is this correct?
what are we using now?

Reply:
module arguments is more general i guess, since it could contain other objects, not just hyper parameters.

Reply:
the checkpoint shouldn't contain objects no?
i think the user has to pass those in specifically
```
GAN.load_from_checkpoint(PATH, generator=Gen(), disc=D())
```

Reply:
I looks like checkpoint will contain the pickled state dict for GAN and separately state dicts for Gen and D, which makes the checkpoint weight nearly twice as large as should be.

Reply:
given the description of the PR and what was discussed, I had the impression that hparams are gone and now we just consider module arguments and save them, as long as they are pickleable, and if not, we would warn the user. 
as @festeh pointed out, this could be too heavy https://github.com/PyTorchLightning/pytorch-lightning/pull/1896#discussion_r429616639

the name "module_arguments" would be a misleading name if this wasn't the case.

Reply:
When doing auto save of args there are two ways:
* Save only primitives like int, list, dict... (by checking __dict__ presence) -> that won't allow saving custom config classes
* Save everything that is picklable -> that may store some redundant and possibly heavy state with each checkpoint

Reply:
the point is not to save objects. Only primitives.

Reply:
this API stays the same...

Reply:
ok, @Borda confirm that we are not saving objects? only primivites, lists, dicts and tuples?

Reply:
Changed now